src/backend/joanie/tests/core/test_api_order.py

```diff
@@ -866,7 +977,19 @@ def test_api_order_create_has_read_only_fields(self):
             {
                 "id": str(order.id),
                 "certificate": None,
-                "course": course.code,
+                "course": {
+                    "code": course.code,
+                    "id": str(course.id),
+                    "title": course.title,
+                    "cover": {
+                        "size": course.cover.size,
+                        "src": f"http://testserver{course.cover.url}.1x1_q85.webp",
+                        "srcset": build_srcset_for_field(course.cover),
+                        "height": course.cover.height,
+                        "width": course.cover.width,
+                        "filename": course.cover.name,
```

Comment:
this looks very painful to test and is repeated many times. I think I would like to test it only once and mock it elsewhere :sweat_smile: